toolkit/tools/imagepkgfetcher/imagepkgfetcher.go

```diff
@@ -64,21 +64,12 @@ func main() {
 
 	timestamp.StartEvent("initialize and configure cloner", nil)
 
-	cloner := rpmrepocloner.New()
-	err := cloner.Initialize(*outDir, *tmpDir, *workertar, *existingRpmDir, *existingToolchainRpmDir, *usePreviewRepo, *repoFiles)
+	cloner, err := rpmrepocloner.ConstrcuctClonerWithNetwork(*outDir, *tmpDir, *workertar, *existingRpmDir, *existingToolchainRpmDir, *tlsClientCert, *tlsClientKey, *usePreviewRepo, *disableUpstreamRepos, *repoFiles)
```

Comment:
When you pass `disableUpstreamRepo` as argument, the function name can be just `ConstructCloner`. Because based on `disableUpstreamRepo` value, the cloner may not have networking capability.

Reply:
I've renamed to `ConstructCloner()`, and given it a slight update to have `disableNetwork` as a parameter instead of `disableUpstreamRepos` which is much more descriptive.